fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/ListPartitionPruner.java

```diff
@@ -183,17 +189,21 @@ public List<Long> prune() throws AnalysisException {
         }
     }
 
-    private Set<Long> evalPartitionPruneFilter(ScalarOperator operator) {
+    private Pair<Set<Long>, Boolean> evalPartitionPruneFilter(ScalarOperator operator) {
+        Set<Long> matches = null;
+        Boolean existNoEval = false;
         if (operator instanceof BinaryPredicateOperator) {
-            return evalBinaryPredicate((BinaryPredicateOperator) operator);
+            matches = evalBinaryPredicate((BinaryPredicateOperator) operator);
         } else if (operator instanceof InPredicateOperator) {
-            return evalInPredicate((InPredicateOperator) operator);
+            matches = evalInPredicate((InPredicateOperator) operator);
         } else if (operator instanceof IsNullPredicateOperator) {
-            return evalIsNullPredicate((IsNullPredicateOperator) operator);
+            matches = evalIsNullPredicate((IsNullPredicateOperator) operator);
         } else if (operator instanceof CompoundPredicateOperator) {
-            return evalCompoundPredicate((CompoundPredicateOperator) operator);
+            Pair<Set<Long>, Boolean> matchesPair = evalCompoundPredicate((CompoundPredicateOperator) operator);
+            matches = matchesPair.first;
+            existNoEval = matchesPair.second;
         }
-        return null;
+        return matches == null ? Pair.create(null, true) : Pair.create(matches, existNoEval);
```

Comment:
nit: it's best to use empty to replace null.

Reply:
https://github.com/StarRocks/starrocks/blob/257281c4f17caeee1371d3451fee9fcc08a9dc01/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/ListPartitionPruner.java#L156C24-L156C24
Here it will be judged whether it is `null` instead of `empty`.